### PR TITLE
Fix caching in Python 3.6/Windows

### PR DESCRIPTION
**Pull request by tarkatronic**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [X] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [X] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [X] Yes (backward compatible)
* [ ] No (breaking changes)

## What's new?

Attempting to fix caching in Windows CI. If this doesn't work, we'll just go ahead and disable caching for Windows.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,9 +36,10 @@ jobs:
         path: |
           {{ matrix.path }}
           .tox
-        key: ${{ runner.os }}-venvs-${{ hashFiles('**/poetry.lock') }}
+        key: ${{ matrix.os }}-venvs-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
         restore-keys: |
-          ${{ runner.os }}-venvs
+          ${{ matrix.os }}-venvs-${{ matrix.python-version }}
+          ${{ matrix.os }}-venvs
```
Comment (rbailey-godaddy) [inline .github/workflows/ci.yml:43]: So I barely understand what this is doing, although I am reverse-engineering it in my mind and understand the intent.

With that proviso, should the order of L41 and L42 be reversed? I am assuming that in the case of key conflicts, the later definition would overwrite an earlier one, and presumably we would want any python-version-specific definitions to replace more generic ones...

Basically clueless here.

**Review by rbailey-godaddy — COMMENTED** [review]
I provided some feedback, but I am not competent to assess the correctness of the change, so I'll leave that to somebody else unless we are tooooootally desperate. ;-)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tarkatronic) [reply .github/workflows/ci.yml:43]: Nope! 😄 
https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action

**Comment by tarkatronic** [comment]
In the end, I went with skipping the caching on Python 3.6/Windows. `tox` was failing with an exit code of `3221225504`, which translates to `0xC0000005`, or `STATUS_ACCESS_VIOLATION`: `The instruction at 0x%08lx referenced memory at 0x%08lx. The memory could not be %s.`

Reference: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55?redirectedfrom=MSDN

TL;DR: Caching is broken in Python 3.6/Windows, so we just don't do it now.

**Review by rbailey-godaddy — APPROVED** [review]
I approve, but I learned most of what I know in this area from @jwilhelm-godaddy while reviewing this PR... ;-) But it is nice to see all of the automated tests passing again now.

**Review by sushantmimani — APPROVED** [review]
LGTM 🦅
